Library/Homebrew/utils.sh

```diff
@@ -0,0 +1,47 @@
+puts_with_style() {
+  local message output_fd="$1" plain="$2" styled="$3"
+  shift 3
+
+  if [[ $# -eq 0 ]]
+  then
+    IFS= read -d '' -r message # Read message from `stdin`.
+    message="${message%$'\n'}" # Trim trailing newline.
+  else
+    message="$*"
+  fi
+
+  if [[ -t "$output_fd" ]] # If FD is a TTY, style message with color/bold/...
+  then
+    # shellcheck disable=SC2059
```

Comment:
Yes, I can put that explanation into the code. We can't use `echo` here because the message goes in the middle of a format string that (for the typical case of a TTY) contains escape sequences for styling before and/or after the `%s` placeholder.